.github/workflows/release.yml

```diff
@@ -0,0 +1,270 @@
+      - name: Is Pre-release
+        id: is_prerelease
+        run: |
+          echo ${{ github.ref }} | grep "\-rc.*"
+          OUT=$?
+          if [ $OUT -eq 0 ]; then IS_PRERELEASE=true; else IS_PRERELEASE=false; fi
+          echo "::set-output name=IS_PRERELEASE::${IS_PRERELEASE}"
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: Release ${{ github.ref }}
+          draft: false
+          prerelease: ${{ steps.is_prerelease.outputs.IS_PRERELEASE }}
+
+  build_upload_github_assets:
+    name: Build & Upload Mac OS Darwin Assets To Github Release
+    runs-on: ubuntu-latest
+    needs: create_release
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15.6'
+      - name: Build
+        id: build
+        run: |
+          PATH=$PATH:/usr/local/go/bin:/home/admin/go/bin
+          make -j1 darwin-artifacts linux-artifacts windows-artifacts
+      - name: Upload Release Assets
+        id: upload-release-assets
+        run: |
+          tag_name="${GITHUB_REF##*/}"
+          hub release edit $(find ./.releases -type f -printf "-a %p ") -m "" "$tag_name"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+# All jobs below this are for full releases (non release candidates e.g. *-rc.*)
+
+  build_upload_docker:
+    name: Build & Upload Docker Images
+    runs-on: ubuntu-latest
+    needs: create_release
+    if: ${{ needs.create_release.outputs.is_prerelease }} == 'false'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15.6'
+      - name: Build
+        id: build
+        run: |
+          PATH=$PATH:/usr/local/go/bin:/home/admin/go/bin
+          make docker-artifacts
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+
+  build_upload_aws_s3_binaries:
+    name: Build & Upload AWS S3 Binaries
+    runs-on: ubuntu-latest
+    needs: create_release
+    if: ${{ needs.create_release.outputs.is_prerelease }} == 'false'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15.6'
+      - name: Build
+        id: build
+        run: |
+          PATH=$PATH:/usr/local/go/bin:/home/admin/go/bin
+          make -j1 binary-linux binary-darwin binary-windows
+          mkdir -p ./.releases
+          cp ./output/binary/linux/bin/step ./.releases/step-linux-${{ needs.create_release.outputs.version }}
+          cp ./output/binary/linux/bin/step ./.releases/step-linux-latest-integration
```

Comment:
What is this? Do we need it?

Reply:
Are you asking if we need the `step-linux-latest-integration`? That's the script used by the integration tests. Rather than specifying a version, they always pull `latest`.